pkg/object/file.go

```diff
@@ -318,7 +321,42 @@ func readDirSorted(dirname string) ([]*mEntry, error) {
 }
 
 func (d *filestore) List(prefix, marker, delimiter string, limit int64) ([]Object, error) {
-	return nil, notSupported
+	if delimiter != "/" {
+		return nil, notSupported
+	}
+	var dir string = d.root + prefix
+	if !strings.HasSuffix(dir, dirSuffix) {
+		dir = path.Dir(dir) + dirSuffix
+	}
+	entries, err := readDirSorted(dir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+	var objs []Object
+	for _, e := range entries {
+		p := filepath.Join(dir, e.Name())
+		if e.IsDir() {
+			p = filepath.ToSlash(p + "/")
+		}
+		key := p[len(d.root):]
```

Comment:
it will panic if `len(d.root) > len(dir) + len(name)`.
e.g. List `/a/bb`, while there's another entry named `/a/b`